server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java

```diff
@@ -220,4 +274,130 @@ public void writeTo(StreamOutput out) throws IOException {
+            );
+
+            static {
+                PARSER.declareString(
+                    ConstructingObjectParser.optionalConstructorArg(),
+                    value -> TimeValue.parseTimeValue(value, AFTER_FIELD.getPreferredName()),
+                    AFTER_FIELD
+                );
+                PARSER.declareString(
+                    ConstructingObjectParser.optionalConstructorArg(),
+                    value -> TimeValue.parseTimeValue(value, FIXED_INTERVAL_FIELD.getPreferredName()),
+                    FIXED_INTERVAL_FIELD
+                );
+
+            }
+
+            public static Round read(StreamInput in) throws IOException {
+                return new Round(in.readTimeValue(), in.readTimeValue());
+            }
+
+            @Override
+            public void writeTo(StreamOutput out) throws IOException {
+                out.writeTimeValue(after);
+                out.writeTimeValue(fixedInterval);
+            }
+
+            @Override
+            public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+                builder.startObject();
+                builder.field(AFTER_FIELD.getPreferredName(), after.getStringRep());
+                builder.field(FIXED_INTERVAL_FIELD.getPreferredName(), fixedInterval.getStringRep());
+                builder.endObject();
+                return builder;
+            }
+
+            public static Round fromXContent(XContentParser parser, Void context) throws IOException {
+                return PARSER.parse(parser, context);
+            }
+
+            @Override
+            public String toString() {
+                return Strings.toString(this, true, true);
+            }
+        }
+
+        // For testing
+        public static final Downsampling NULL = new Downsampling(null);
+
+        public Downsampling {
+            if (rounds != null) {
+                if (rounds.isEmpty()) {
+                    throw new IllegalArgumentException("Downsampling configuration should have at least one round configured.");
+                }
+                Round previous = null;
+                for (Round round : rounds) {
+                    if (previous == null) {
+                        previous = round;
+                    } else {
+                        if (round.after.compareTo(previous.after) < 0) {
+                            throw new IllegalArgumentException(
+                                "A downsampling round must have a later 'after' value than the proceeding, "
+                                    + round.after.getStringRep()
+                                    + " is not after "
+                                    + previous.after.getStringRep()
+                                    + "."
+                            );
+                        }
+                        if (round.fixedInterval.compareTo(previous.fixedInterval) < 0) {
```

Comment:
Shall we also introduce the validation that all  fixed_intervals are required to be multiples of each other? 

(This could be a follow-up as well)

Reply:
I will give it a try.

Reply:
I abstracted this validation so now it's shared.